src/wallet/components/Wallet.test.tsx

```diff
@@ -29,45 +29,33 @@ vi.mock('../../ui/react/internal/hooks/useOutsideClick', () => ({
   useOutsideClick: vi.fn(),
 }));
 
-type useOutsideClickType = ReturnType<
-  typeof vi.fn<
-    (
-      ref: React.RefObject<HTMLElement>,
-      callback: (event: MouseEvent) => void,
-    ) => void
-  >
->;
-
 describe('Wallet Component', () => {
-  let mockSetIsOpen: ReturnType<typeof vi.fn>;
-  let mockOutsideClickCallback: (e: MouseEvent) => void;
+  let mockHandleClose: ReturnType<typeof vi.fn>;
 
   beforeEach(() => {
-    mockSetIsOpen = vi.fn();
+    mockHandleClose = vi.fn();
     (useWalletContext as ReturnType<typeof vi.fn>).mockReturnValue({
       isOpen: true,
-      setIsOpen: mockSetIsOpen,
+      handleClose: mockHandleClose,
+      containerRef: { current: document.createElement('div') },
     });
 
-    (useOutsideClick as unknown as useOutsideClickType).mockImplementation(
-      (_, callback) => {
-        mockOutsideClickCallback = callback;
-      },
-    );
-
     vi.clearAllMocks();
   });
 
   it('should render the Wallet component with ConnectWallet', () => {
     (useWalletContext as ReturnType<typeof vi.fn>).mockReturnValue({
       isOpen: false,
-      setIsOpen: mockSetIsOpen,
+      handleClose: mockHandleClose,
+      containerRef: { current: document.createElement('div') },
     });
 
     render(
       <Wallet>
-        <ConnectWallet />
-        <WalletDropdown />
+        <WalletBasic>
+          <ConnectWallet />
+          <WalletDropdown />
+        </WalletBasic>
```

Comment:
won't wallet island and and current share the connect wallet components?  would't the only change be <WalletDropdown/> vs. <WalletIsland/>?

Reply:
yes, that's correct. The reason for this change is that WalletBasic is opinionated about children, destructuring and reorganizing them.

Reply:
is it possible to use a single <Wallet/> top level node and then look for WalletDropdown or WalletIsland to determine which?